src/Operation/Find.php

```diff
@@ -418,6 +419,16 @@ private function createQueryOptions(): array
             $options['modifiers'] = is_object($modifiers) ? document_to_array($modifiers) : $modifiers;
         }
 
+        // Ensure no cursor is left behind when limit == batchSize by increasing batchSize
+        if (isset($options['limit'], $options['batchSize']) && $options['limit'] === $options['batchSize']) {
+            assert(is_integer($options['batchSize']));
+            $options['batchSize']++;
+        }
+
+        if (isset($options['limit']) && $options['limit'] === 1) {
+            $options['singleBatch'] = true;
```

Comment:
> I took the step of automatically setting singleBatch when limit is set to 1, even though this can affect find operations where users themself set limit to 1. Let me know if you see any potential issues with that approach, the alternative would be to introduce singleBatch as an undocumented option to Find.

@alcaeus: Is that any different than what already happens in PHPC? Read on.

----

The CRUD spec's Q&A says this about why `singleBatch` was omitted from FindOptions in the API:

> Rather than introduce a singleBatch option to FindOptions, the spec preserves the existing API for limit and instructs drivers to convert negative values accordingly for servers >= 3.2.

That said, the option [exists in PHPC](https://www.php.net/manual/en/mongodb-driver-query.construct.php). PHPC also supports a negative limit, which implies `singleBatch: true`. That logic is implemented in [`php_phongo_query_init_limit_and_singlebatch`](https://github.com/mongodb/mongo-php-driver/blob/1.20.0/src/MongoDB/Query.c#L201).

I think there are two possible actions to take here:

 * Acknowledge the CRUD spec and leave support for negative limits in place. If so, we should update the docs.
 * Deprecate negative limits in 1.x and remove it in 2.0 (to be on par with libmongoc's API). In that case we should introduce `singleBatch` as an explicit option in PHPLIB.

Reply:
Yeah, I believe given that we no longer support 3.2 and older, we may want to consider introducing `singleBatch` as an option for `Find`. Reading the spec, it never says that a driver may not provide `singleBatch` as an option, so we could certainly add it for PHP. We can always decide to leave it undocumented until the spec is updated. I can file a drivers ticket for adding `singleBatch`.

That said, `singleBatch` has scenarios where not all expected results are returned (so neither `limit` nor `batchSize` is reached). This is not relevant for `findOne`, but I feel like that should be documented in the spec.